### PR TITLE
Download model files in parallel when < X files

### PR DESCRIPTION
**Pull request by rosbo**
Blocked by: https://github.com/Kaggle/kaggleazure/pull/29608 (@stevemessick is working on the backend fix)

http://b/341160276

### CODE REVIEW

**Comment by rosbo** [comment]
@jeward414 I will add tests on Monday but here is what we did together. I also added the parallel download part.

**Comment by rosbo** [comment]
Note: integration test is failing until https://github.com/Kaggle/kaggleazure/pull/29608 is merged (@stevemessick is working on the backend fix). I will wait for the change to be deployed before merging this PR.

**Review by jeward414 — APPROVED** [review]

File: src/kagglehub/http_resolver.py [hunk 1]
```diff
@@ -80,10 +102,27 @@ def _get_current_version(api_client: KaggleApiV1Client, h: ModelHandle) -> int:
 
     return json_response[MODEL_INSTANCE_VERSION_FIELD]
 
+def _list_files(api_client: KaggleApiV1Client, h: ModelHandle) -> tuple[list[str], bool]:
+    json_response = api_client.get(_build_list_model_instance_version_files_url_path(h), h)
```
[the remark this answers is not in the document]
Comment (rosbo) [reply src/kagglehub/http_resolver.py:106]: The fix for the API has been released to prod.

**Comment by rosbo** [comment]
I found why the integration tests were failing on CI but not when running locally.

There is a `kaggle-api-enable-pagination` feature flag for the new ListModelInstanceVersionFilesHandler. It is currently released to `50%`. The `integrationtester` on CI didn't fall into this bucket so it was always getting an empty response when listing files: https://github.com/Kaggle/kaggleazure/blob/dc89139c44f4d0ab33c1331c4326340619b3f6a0/Kaggle.Services.Models/Handlers/ModelApiService/V1/ListModelInstanceVersionFilesHandler.cs#L86

@stevemessick Are you planning to turn the feature flag to `public` soon? Also, did we need the feature flag for the model handler given it is new?

**Comment by stevemessick** [comment]
@rosbo That FF was supposed to have been 100% but the update didn't stick. I just verified it is now 100%.

The FF will be removed by https://github.com/Kaggle/kaggleazure/pull/29772.

**Comment by stevemessick** [comment]
I re-ran the failing check, and it is passing now.

**Pull request merged**